php/libraries/ICandidate.class.inc

```diff
@@ -0,0 +1,319 @@
+    /**
+     * Get a list of SessionIDs for which this Candidate has a corresponding
+     * TimePoint.
+     *
+     * @return SessionID[]
+     */
+    public function getListOfTimePoints(): array;
+
+    /**
+     * Returns the sessionID for this Candidate's nth visit
+     *
+     * @param integer $visitNo The Nth visit for a Candidate.
+     *
+     * @return ?SessionID SessionID for visit specified by $visitNo
+     */
+    public function getSessionID(int $visitNo): ?SessionID;
+
+    /**
+     * Manages the parameter_candidate tableso that projects have a way to add
+     * whatever data they want which is outside the scope of LORIS to their
+     * candidates.
+     *
+     * @param string|null $var The type of data about the Candidate
+     *                    to be returned
+     *
+     * @return     mixed The info requested
+     * @deprecated
+     */
+    public function getData(?string $var = null);
+
+
+    /**
+     * Returns the ProjectID for the current candidate
+     *
+     * @return integer
+     */
+    public function getProjectID(): int;
+
+    /**
+     * Returns the name of the candidate's project from the
+     * config file
+     *
+     * @return string Project name
+     */
+    public function getProjectTitle(): string;
+
+    /**
+     * Returns the CandID of this candidate
+     *
+     * @return CandID 6 digit CandID
+     */
+    public function getCandID();
+
+    /**
+     * Returns the PSCID of this candidate
+     *
+     * @return string The PSCID identifier for candidate
+     */
+    public function getPSCID(): string;
+
+    /**
+     * Returns the site of this candidate
+     *
+     * @return Site
+     */
+    public function getCandidateSite(): Site;
+
+
+    /**
+     * Returns the ID of this candidate's site
+     *
+     * @return int The numeric ID of the candidate's site
+     */
+    public function getCenterID(): int;
+
+    /**
+     * Returns the date of birth of this candidate
+     *
+     * @return ?DateTime
+     */
+    public function getCandidateDoB(): ?DateTime;
+
+    /**
+     * Returns the expect date of confinement (due date)
+     * of this candidate.
+     *
+     * @return ?DateTime
+     */
+    public function getCandidateEDC(): ?DateTime;
+
+    /**
+     * Return the candidate's sex
+     *
+     * @return string|null Male|Female String describing the sex
+     *                     and null if scanner
+     */
+    public function getCandidateSex(): ?string;
+
+    /**
+     * Return whether the candidate is active
+     *
+     * @return bool
+     */
+    public function isActive(): bool;
+
+    /**
+     * Return the user that registered (created) this candidate
+     *
+     * @return string UserID of user who created this candidate.
+     */
+    public function getRegisteredBy(): string;
+
+    /**
+     * Return the name of the user account that last changed this candidate.
+     *
+     * @return User
+     */
+    public function getLastModifiedBy(): User;
```

Comment:
They're fields in the candidate table that are used by the current class.

It can be removed from the interface but this is probably a sort of "code smell" about the structure of the table

Reply:
Being in the candidate table doesn't mean it needs to be in a candidate interface.. if needed, I think, something like "getLastModified(By)" should make more sense in some kind of generic data provenence interface than directly in the candidate interface.